lib/diagnostic.rb

```diff
@@ -13,28 +19,35 @@ def longest(enum)
   # enumerable supplied (i.e. apply the `+` operator)
   # Assume each entry implements a `+` method
   def sum(enum)
-    # Your code goes here
+    enum.reduce(0) { |m, p| m += p }
   end
 
   # Question 3
   # Write ruby code that returns true if some of the entries
   # in the numbers enumerable are odd
   def some_odd(numbers)
-    # Your code goes here
+    numbers.each do |num|
+      if num.odd?
+        return true
+      end
+    end
   end
 
   # Question 4
   # Write ruby code that returns true if every entry in the
   # numbers enumerable are even
   def every_even(numbers)
-    # Your code goes here
+    sum = numbers.reduce(0) { |memo, obj| memo += obj }
+    sum % 2 == 0
   end
 
   # Question 5
   # Write ruby code that returns an array with all of the
   # entries from the words enumerable capitalized (i.e. 'foo' becomes 'Foo').
   def transform(words)
-    # Your code goes here
+    words.each do |word|
+      word.capitalize!
+    end
```

Comment:
This will work. A better approach, though, would be to use `.map`.

`words.map { |word| word.capitalize }` ( or `words.map(&:capitalize)`)